model/src/pyrenew/process/periodiceffect.py

```diff
@@ -0,0 +1,161 @@
+# numpydoc ignore=GL08
+
+from typing import NamedTuple
+
+import jax.numpy as jnp
+import pyrenew.arrayutils as au
+from pyrenew.metaclass import RandomVariable, _assert_sample_and_rtype
+
+
+class PeriodicEffectSample(NamedTuple):
+    """
+    A container for holding the output from
+    `process.PeriodicEffect.sample()`.
+
+    Attributes
+    ----------
+    value: jnp.ndarray
+        The sampled value.
+    """
+
+    value: jnp.ndarray
+
+    def __repr__(self):
+        return f"PeriodicEffectSample(value={self.value})"
+
+
+class PeriodicEffect(RandomVariable):
```

Comment:
I think we can get rid of the `PeriodicEffect` or we should create an analogous class for `broadcast_type="repeat"`.